### ISSUE TITLE
OpenAPI v3: Form parameters not parsed from requestBody

### ISSUE THREAD

**Issue by gazab** [ferdikoomen/openapi-typescript-codegen#257]
One breaking change of OpenAPI 3.0 is that form and body parameters are both described using requestBody instead of using the old parameter with formdata as in value. See https://swagger.io/docs/specification/describing-request-body/ and https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#requestBodyObject

Unfortunately this doesn't work with this library since it still uses the old way 2.0 of looking for form data among the parameter list. See: https://github.com/ferdikoomen/openapi-typescript-codegen/blob/d95de6ef46fd05b435ede71fecf5e2ba331a48d6/src/openApi/v3/parser/getOperationParameters.ts#L41 
So generated clients just passes the data in the body.

Instead, it should take form parameters from request body if it uses the content type `multipart/form-data` or `application/x-www-form-urlencoded`.

Here are two examples where the OpenAPI spec describes form data that does not work:
Binary form content under the "file" name:
```json
"requestBody": {
          "content": {
            "multipart/form-data": {
              "schema": {
                "type": "object",
                "properties": {
                  "file": {
                    "type": "string",
                    "format": "binary",
                    "nullable": true
                  }
                }
              },
              "encoding": {
                "file": {
                  "style": "form"
                }
              }
            }
          }
        }
```

(This is from my own swagger.json and the reason I write this 😄 )

A pet form with two string string values under the names "name" and "status":
```json
"requestBody": {
    "content": {
      "application/x-www-form-urlencoded": {
        "schema": {
          "type": "object",
           "properties": {
              "name": { 
                "description": "Updated name of the pet",
                "type": "string"
              },
              "status": {
                "description": "Updated status of the pet",
                "type": "string"
             }
           },
        "required": ["status"] 
        }
      }
    }
```
(Taken directly from the OpenAPI 3.0 spec file: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#requestBodyObject)

I think my conclusions above are correct but you have way better understanding of the OpenAPI spec and this library than I do so don't take it for a fact 😄

**Comment by ferdikoomen** [comment]
@gazab Yes, i see the issue. Let me see if there is a nice way to implement this, but the documentation is pretty clear.

**Comment by tomohisa** [comment]
https://github.com/tomohisa/openapi-typescript-codegen
I fixed it for my own project but not sure its the way it should fixed.
https://github.com/ferdikoomen/openapi-typescript-codegen/compare/master...tomohisa:master
I added content which was not used current and depends on content, and add
operation.parametersFormData
to use for
formData: formData
{formData?: any}

**Comment by knoefel** [comment]
Any news on when this will be available as I'm facing the same issue right now? Thx!!
